File: Introduce_Assertion.py
# After
def getExpenseLimit(self):
    # should have either expense limit or a primary project
    return self.expenseLimit if self.expenseLimit != NULL_EXPENSE else self.primaryProject.getMemberExpenseLimit()
        
# Refactoring
def getExpenseLimit(self):
    assert (self.expenseLimit != NULL_EXPENSE) or (self.primaryProject != None)
    return self.expenseLimit if (self.expenseLimit != NULL_EXPENSE) else self.primaryProject.getMemberExpenseLimit();

File: Introduce_Assertion.js
// After
function getExpenseLimit(self) {
  // should have either expense limit or a primary project
  return  (self.expenseLimit != NULL_EXPENSE) ? self.expenseLimit : self.primaryProject.getMemberExpenseLimit();
}    
        
// Refactoring
function getExpenseLimit(self) {
  if ((self.expenseLimit != NULL_EXPENSE) || (self.primaryProject != None)) {
    throw new Error('Should have either expense limit or a primary project');
  }
  return (self.expenseLimit != NULL_EXPENSE) ? self.expenseLimit : self.primaryProject.getMemberExpenseLimit();
}    
